### ISSUE TITLE
Detaching fails if wpa_supplicant killed

### ISSUE THREAD

**Issue by rofrankel** [neculaesei/pywpactrl#1]
If I kill wpa_supplicant (removing the control socket) the WPACtrl object cannot detach, raising a wpactrl.error.  This happens even when the object is garbage collected.  This leads to code like:

    try:
      self.wpa_ctrl.detach()
    except wpactrl.error:
      try:
        del self.wpa_ctrl
      except wpactrl.error:
        pass
      self.wpa_ctrl = None

Do you have a better suggestion for how to handle this case?
